common/phal_F4/rcc/rcc.c

```diff
@@ -0,0 +1,366 @@
+//         flash_acr_temp |= FLASH_ACR_LATENCY_2WS << FLASH_ACR_LATENCY_Pos;
+//     else if (system_clock_target_hz >= 32000000)
+//         flash_acr_temp |= FLASH_ACR_LATENCY_1WS << FLASH_ACR_LATENCY_Pos;
+//     else if (system_clock_target_hz >= 16000000)
+//         flash_acr_temp |= FLASH_ACR_LATENCY_0WS << FLASH_ACR_LATENCY_Pos;
+
+//     FLASH->ACR = flash_acr_temp;
+
+//     __DSB();
+//     RCC->CFGR |= RCC_CFGR_SW_PLL;
+//     while((RCC->CFGR & RCC_CFGR_SWS_PLL != RCC_CFGR_SWS_PLL))
+//         ; // Wait for PLL to be the new system clock
+//     __DSB();
+
+//     SystemCoreClockUpdate();
+//     return true;
+// }
+
+bool PHAL_configureHSISystemClock()
+{
+    // Enable and wait for HSI to enable
+    RCC->CR |= RCC_CR_HSION;
+    while (!(RCC->CR & RCC_CR_HSIRDY))
+        ;
+
+    // __DSB();
+    // PWR->CR1 |= (PWR_CR1_VOS_0 << PWR_CR1_VOS_Pos) & PWR_CR1_VOS_Msk;
+    // while (PWR->SR2 & PWR_SR2_VOSF)
+    //     ; // Wait for flag to clear
+    // __DSB();
+
+    // Flash latency adjustment, see ST RM0395 Table 9
+    uint32_t flash_acr_temp = FLASH->ACR;
+    flash_acr_temp &= ~(FLASH_ACR_LATENCY_Msk);
+    flash_acr_temp |= FLASH_ACR_LATENCY_0WS << FLASH_ACR_LATENCY_Pos;
+    FLASH->ACR = flash_acr_temp;
+
+    __DSB();
+    RCC->CFGR |= RCC_CFGR_SW_HSI;
+    while((RCC->CFGR & RCC_CFGR_SWS_HSI) != RCC_CFGR_SWS_HSI)
+        ; // Wait for HSI to be the new system clock
+    __DSB();
+
+    SystemCoreClockUpdate();
+    return true;
+}
+
+bool PHAL_configureAHBClock(uint32_t ahb_clock_target_hz)
+{
+    uint32_t desired_psc = SystemCoreClock / ahb_clock_target_hz;
+    uint32_t sys_clk_div;
+    switch(desired_psc) // ST RM0394 pg. 198
+    {
+        case 1:
+            sys_clk_div = RCC_CFGR_HPRE_DIV1;
+            break;
+        case 2:
+            sys_clk_div = RCC_CFGR_HPRE_DIV2;
+            break;
+        case 4:
+            sys_clk_div = RCC_CFGR_HPRE_DIV4;
+            break;
+        case 8:
+            sys_clk_div = RCC_CFGR_HPRE_DIV8;
+            break;
+        case 16:
+            sys_clk_div = RCC_CFGR_HPRE_DIV16;
+            break;
+        case 64:
+            sys_clk_div = RCC_CFGR_HPRE_DIV64;
+            break;
+        case 128:
+            sys_clk_div = RCC_CFGR_HPRE_DIV128;
+            break;
+        case 256:
+            sys_clk_div = RCC_CFGR_HPRE_DIV256;
+            break;
+        case 512:
+            sys_clk_div = RCC_CFGR_HPRE_DIV512;
+            break;
+        default:
+            return false; // Invalid prescalar value
+    }
+
+    uint32_t rcc_cfgr_temp = RCC->CFGR;
+    rcc_cfgr_temp &= ~(RCC_CFGR_HPRE_Msk);
+    rcc_cfgr_temp |= (sys_clk_div);
+
+    RCC->CFGR = rcc_cfgr_temp;
+
+    AHBClockRateHz = ahb_clock_target_hz;
+    return true;
+}
+
+bool PHAL_configureAPB1Clock(uint32_t apb1_clock_target_hz)
+{
+    uint32_t desired_psc = AHBClockRateHz / apb1_clock_target_hz;
+    uint32_t ahb_clk_div;
+    switch(desired_psc) // ST RM0394 pg. 198
```

Comment:
r these clock configs same on f4 as l4?

Reply:
There will be differences - RCC HAL currently being developed, so this is a placeholder that works